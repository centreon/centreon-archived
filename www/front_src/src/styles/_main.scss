@import "partials/typography";
@import "partials/fonts";
@import "partials/variables";
@import "partials/messages";
@import "partials/icons";
@import "partials/form/form";
@import "partials/helpers";
@import "partials/mixins";
