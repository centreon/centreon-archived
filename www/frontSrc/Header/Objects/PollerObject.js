import React from 'react'
import Button from 'material-ui/Button'
import Grid from 'material-ui/Grid'
import { withStyles } from 'material-ui/styles'
import Popover from 'material-ui/Popover'
import Typography from 'material-ui/Typography'
import PollerIcon from "../icons/PollerIcon"

const styles = theme => ({
  root: {
    position: 'relative',
    fontFamily: theme.font.openSans,
    display: 'inline-flex',
    verticalAlign: 'middle',
    margin: '6px',
  },
  'a': {
    color: '#0072CE',
    '&:hover': {
      backgroundColor: '#064166'
    },
    '&:visited': {
      color: '#10069F'
    }
  },
  status: {
    margin: '4px',
    color: '#fff',
    width: 38,
    height: 38,
    '& span': {
      fontSize: 16
    },
  },
  errorStatus: {
    margin: '10px 4px',
    width: 46,
    height: 46,
    backgroundColor: theme.palette.error.main,
    '& span': {
      fontSize: 16,
      color: '#fff',
      fontWeight: '600'
    },
  },
  chip: {
    height: '8px',
    width: '8px',
    borderRadius: 20,
    display: 'inline-table',
    marginRight: 6,
  },
  icon: {
    width: 34,
    height: 34,
    cursor: 'pointer',
  },
  paper: {
    padding: theme.spacing.unit,
  },
  popover: {
    pointerEvents: 'none',
  },
  objectDetails: {
    padding: '10px 16px',
    borderBottom: '1px solid #d1d2d4',
    '&:last-child' : {
      borderBottom: 'none',
    }
  },
  total: {
    float: 'right',
    marginLeft: 34,
  },
  errorNotif: {
    backgroundColor: theme.palette.error.lighter,
    color: theme.palette.error.main,
    padding: 10,
    borderRadius: 3,
  },
  warningNotif: {
    ackgroundColor: theme.palette.warning.light,
    color: theme.palette.warning.main,
    padding: 10,
    borderRadius: 3,
    textAlign: 'center',
  },
  primaryButton: {
    textTransform: 'initial',
    border: '1px solid ' + theme.palette.primary.dark,
    color: theme.palette.primary.main,
    '&:hover': {
      color: '#FFF',
      backgroundColor: theme.palette.primary.main,
    }
  },
})

const PollerObject = (
  {classes, iconColor, message, total, anchorEl, open, handleClose, handleOpen, className,
    database, latency, stability
  }) => (
  <div className={classes.root}>
    <PollerIcon
      id='pollerIcon'
      alt="poller icon"
      aria-haspopup="true"
      onClick={handleOpen}
      viewBox="6 156 600 600"
      className={classes.icon}
      nativeColor={iconColor}
    />
    <Popover
      id='pollerPopover'
      open={open}
      anchorEl={anchorEl}
      anchorReference='anchorEl'
      anchorPosition={{ top: 500, left: 400 }}
      onClose={handleClose}
      anchorOrigin={{
        vertical: 'bottom',
        horizontal: 'left',
      }}
      transformOrigin={{
        vertical: 'top',
        horizontal: 'left',
      }}
    >
      <div className={classes.objectDetails}>
        <Typography variant="caption" gutterBottom>
            All pollers
          <span className={classes.total}>{total}</span>
        </Typography>
      </div>
      <div className={classes.objectDetails}>
        <Typography variant="caption" gutterBottom>
          <span className={classes.chip} style={{backgroundColor: '#e00b3d'}}></span>
          {stability.critical.message}
          <span className={classes.total}>{stability.critical.total}</span>
        </Typography>
        <Typography variant="caption" gutterBottom>
          <span className={classes.chip} style={{backgroundColor: '#FF9A13'}}></span>
          {stability.warning.message}
          <span className={classes.total}>{stability.warning.total}</span>
        </Typography>
      </div>
      <div className={classes.objectDetails}>
        <Typography variant="caption" gutterBottom>
          <span className={classes.chip} style={{backgroundColor: '#e00b3d'}}></span>
          {latency.critical.message}
          <span className={classes.total}>{latency.critical.total}</span>
        </Typography>
        <Typography variant="caption" gutterBottom>
          <span className={classes.chip} style={{backgroundColor: '#FF9A13'}}></span>
          {latency.warning.message}
          <span className={classes.total}>{latency.warning.total}</span>
        </Typography>
      </div>
      <div className={classes.objectDetails}>
        <Typography variant="caption" gutterBottom>
          <span className={classes.chip} style={{backgroundColor: '#e00b3d'}}></span>
          {database.critical.message}
          <span className={classes.total}>{database.critical.total}</span>
        </Typography>
        <Typography variant="caption" gutterBottom>
          <span className={classes.chip} style={{backgroundColor: '#FF9A13'}}></span>
          {database.warning.message}
          <span className={classes.total}>{database.warning.total}</span>
        </Typography>
      </div>
      <div className={classes.objectDetails}>
        <Button className={classes.primaryButton} href="./main.php?p=609">
          Configure pollers
        </Button>
      </div>
    </Popover>
  </div>
)

export default withStyles(styles)(PollerObject)
