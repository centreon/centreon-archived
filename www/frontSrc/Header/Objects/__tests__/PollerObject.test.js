/* eslint-env jest */
import React from 'react'
import Enzyme from 'enzyme'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
import PollerObject from '../PollerObjectContainer'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { MuiThemeProvider, createMuiTheme } from '@material-ui/core/styles'

Enzyme.configure({ adapter: new Adapter() })

const theme = createMuiTheme({
  palette: {
    primary: {main: '#88b917', dark: '#597F00'},
    secondary: {main: '#00bfb3'},
    error: {main: '#e00b3d', light: '#ff2c5b', lighter: '#FF8C88'},
    warning: {main: '#ff9a13', light: '#ffb749'},
    unreachable: {main: '#818285', light: '#a6a7a6'},
    unknown: {main: '#bcbdc0', dark: '#95969a'},
    pending: {main: '#2AD1D4'},
  },
  font: {
    openSans: "'Open Sans', Arial, Tahoma, Helvetica, Sans-Serif"
  },
})

describe('PollerObject', () => {
  const middlewares = [ thunk ]
  const mockStore = configureStore(middlewares)

  const store = mockStore({
    poller: {
      issues: {
        database: {
          critical: {
            poller: [
              {
                id: 1,
                name: 'central',
                since: 12345
              }
            ],
            total: 1
          },
          warning: {
            total: 0,
            poller: [
              {
                id: 1,
                name: 'central',
                since: 12345
              }
            ],
          },
        },
        total: 12
      },
      refreshTime: 90,
      total: 150,
      dataFetched: true
    }
  })

  let getPollers = jest.fn()

  it('Poller icon renders correctly', () => {
    const component = shallow(<PollerObject store={store} />)
    expect(toJson(component)).toMatchSnapshot()
  })

  it('should display detail popover when clicking on poller icon', () => {
    const component = mount(
      <MuiThemeProvider theme={theme}>
        <PollerObject store={store} getPollers={getPollers} />
      </MuiThemeProvider>
    )

    expect(component.find('div#pollerPopover').length).toEqual(0)
    component.find('svg#pollerIcon').simulate('click')
    expect(component.find('div#pollerPopover').length).toEqual(1)
  })
})