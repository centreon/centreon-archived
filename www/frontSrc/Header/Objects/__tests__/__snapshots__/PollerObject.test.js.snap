// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PollerObject Poller icon renders correctly 1`] = `
<PollerObjectContainer
  getPollersListIssues={[Function]}
  poller={
    Object {
      "dataFetched": true,
      "issues": Object {
        "database": Object {
          "critical": Object {
            "poller": Array [
              Object {
                "id": 1,
                "name": "central",
                "since": 12345,
              },
            ],
            "total": 1,
          },
          "warning": Object {
            "poller": Array [
              Object {
                "id": 1,
                "name": "central",
                "since": 12345,
              },
            ],
            "total": 0,
          },
        },
        "total": 12,
      },
      "refreshTime": 90,
      "total": 150,
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
